set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java

```diff
@@ -362,45 +364,86 @@ String buildSearchApiUrlForClosedSets(UserSet userSet, String apiKey, int pageSi
 	}
 	// close bracket
 	query.append(')');
-	StringBuilder url = new StringBuilder(getConfiguration().getSearchApiUrl());
-	url.append(URLEncoder.encode(query.toString(), StandardCharsets.UTF_8));
-
-	url.append('&').append(CommonApiConstants.PARAM_WSKEY).append('=').append(apiKey);
-	// append rows=100
-	url.append('&').append(CommonApiConstants.QUERY_PARAM_ROWS).append('=').append(maxItems);
+	searchApiRequest.setQuery(query.toString());
```

Comment:
for Closed sets we were setting only these two fields : 
`searchApiRequest.setQuery(query.toString());
searchApiRequest.setRows(maxItems)`. basically we deref all the items for closed sets.
 For closed sets the maxItems/pageSize is : `Math.min(userSet.getTotal(), getConfiguration().getMaxRetrieveDereferencedItems());` and `getMaxRetrieveDereferencedItems `is set to 100

Reply:
done

Reply:
pageNr is not needed here, as by default the start is set to 1 if not passed or set anywhere.

Reply:
I think I remember now, why we didn't use a page here. Because of possible wrong item ids, the result page is built in memory for the closed sets. all items up to pagesize are retrieved here. 

Please leave a comment in the code to have this documented.